src/components/Views/Settings.jsx

```diff
@@ -1,36 +1,38 @@
-import React, { useMemo } from 'react'
+import React from 'react'
 
+import Spinner from 'cozy-ui/transpiled/react/Spinner'
 import SelectBox from 'cozy-ui/transpiled/react/SelectBox'
 import Label from 'cozy-ui/transpiled/react/Label'
 import { useI18n } from 'cozy-ui/transpiled/react/I18n'
-import { useQuery } from 'cozy-client'
 
-import { useAccountContext } from 'src/components/Providers/AccountProvider'
-import { buildAccountQuery } from 'src/queries/queries'
+import {
+  useAccountContext,
+  getAccountLabel
+} from 'src/components/Providers/AccountProvider'
 import Titlebar from 'src/components/Titlebar'
 import CsvExporter from 'src/components/ExportCSV/CsvExporter'
 
 export const Settings = () => {
   const { t } = useI18n()
-  const { selectedAccount, setSelectedAccount } = useAccountContext()
-
-  const accountQuery = buildAccountQuery()
-  const { data } = useQuery(accountQuery.definition, accountQuery.options)
-
-  const accounts = useMemo(() => {
-    if (!data || data.length < 1) {
-      return []
-    }
-    return data.map(account => ({
-      label: account.auth.login,
-      _id: account._id
-    }))
-  }, [data])
-
-  // To know the reason for not passing directly `accounts`, see
-  // https://github.com/cozy/coachCO2/pull/62#discussion_r812929004
-  const accountName = accounts[0]?.label || ''
-  const accountId = accounts[0]?._id || ''
+  const { accounts, account, setAccount } = useAccountContext()
```

Comment:
I am a bit lost when reading this parameter.


When using useAccountContext, I would expect to receive Account / setAccount. But what is accounts???

Reply:
actually, I think the handelChange function should not be inside this component.
=> 
the component should call an external function handling this.
```
 ({ value }) =>
    setAccount(accounts.find(account => account._id === value))
```

Reply:
let's discuss, if i am not clear 🙏

Reply:
> But what is accounts

Ce sont tous les accounts de l'app

> handelChange function should not be inside this component

j'ai pas compris l'intérêt (à part d'eviter la reconstruction à chaque fois, comme le ferait un useCallback) ?